File: apis/src/main/java/com/beat/apis/booking/application/dto/BookingCancelResponse.java
package com.beat.apis.booking.application.dto;


public record BookingCancelResponse(
	long bookingId,
	BookingStatusType bookingStatus
) {
	public static BookingCancelResponse of(
		long bookingId,
		BookingStatusType bookingStatus) {
		return new BookingCancelResponse(
			bookingId,
			bookingStatus);
	}
}

File: apis/src/main/kotlin/com/beat/apis/booking/api/response/BookingCancelResponse.kt
package com.beat.apis.booking.api.response

import com.beat.apis.booking.api.type.BookingStatusType
import com.beat.application.frontoffice.booking.result.BookingCancelResult

@ConsistentCopyVisibility
data class BookingCancelResponse private constructor(
    val bookingId: Long,
    val bookingStatus: BookingStatusType?,
) {
    companion object {
        fun from(result: BookingCancelResult): BookingCancelResponse = BookingCancelResponse(
            bookingId = result.bookingId,
            bookingStatus = result.bookingStatus?.let(BookingStatusType::valueOf),
        )
    }
}
